### ISSUE TITLE
Elastic Search: Some organization rows not indexed?

### ISSUE THREAD

**Issue by DaleMcGrew** [wevote/WeVoteServer#114]
It appears there aren’t any entries for “Planned” in the Elastic Search database. This seems to be a bug, because I can see organizations with “Planned” in their name in the database: https://api.wevoteusa.org/apis/v1/organizationSearch/?csrfmiddlewaretoken=w3Oe9HKyTtw2BRP7GOj9qktsh1GrWaFz&voter_device_id=oZxaqQuLl7bp7pWIfvLquuWRnYKotd69iG5uSRzGFHWt0jR4fXCp90ooTN7l8K5f4Yk4jEoKrJAMOnJlS08JAv7P&organization_name=planned&organization_email=&organization_website=&organization_twitter_handle=

**Comment by josephevans** [comment]
I can't duplicate this -- if I search for "planned" I get back results for Planned Parenthood, etc. I'm using the test script on admin.wevote.me:/home/ubuntu/test_es.py. If I just search for "plan", I don't get anything. If we want results to show for "plan", I think we need to change the ES mapping for that to work. @DaleMcGrew how do you envision the search working? On full terms or partial terms?

ubuntu@ip-172-31-39-39:~$ ./test_es.py planned
Got 39 hits from all index search:
------------- RESULT --------------
_score: 1.9277314
_type: organization
_id: 583
_source: {u'we_vote_id': u'wv02org567', u'state_served_code': u'', u'organization_website,': u'http://www.ppvotescolorado.org/', u'organization_twitter_handle,': u'PPVotesColorado', u'organization_name': u'Planned Parenthood Votes Colorado', u'twitter_description,': u'PPVC promotes pro-choice & pro-family planning policies & advocates for unrestricted ac
cess to reproductive healthcare services & comprehensive sex ed.'}
_index: organizations
------------- RESULT --------------
_score: 1.5842948
_type: organization
_id: 909
_source: {u'we_vote_id': u'wv02org893', u'state_served_code': u'', u'organization_website,': u'http://advocates.ppin.org/', u'organization_twitter_handle,': None, u'organization_
name': u'Planned Parenthood of Indiana', u'twitter_description,': None}
_index: organizations
------------- RESULT --------------
_score: 1.5842948
_type: organization
_id: 316
_source: {u'we_vote_id': u'wv02org300', u'state_served_code': u'', u'organization_website,': u'http://www.plannedparenthoodaction.org/', u'organization_twitter_handle,': u'PPact'
, u'organization_name': u'Planned Parenthood Action Fund', u'twitter_description,': u"We're the Planned Parenthood Action Fund, a 501(c)(4) organization. We engage in education a
nd advocacy to protect and advance women's health and rights."}
_index: organizations
.......

**Comment by DaleMcGrew** [comment]
I would love the return of partial search results, so "plan" returns any item with that text fragment in it. If finding "ann" within "Planned" is too resource intensive, we could start with fragments at the start of words. So we could find "planned" with the "pla" fragment, but not the "ann" fragment.

I have verified that I am getting the same results you are -- as soon as I type in the complete word "planned" I get results too.

**Comment by josephevans** [comment]
@DaleMcGrew Can you change the search query code to be like this:

query = { "query": { "multi_match": { "type": "phrase_prefix", "query": search_term, "fields": [ "candidate_name", "candidate_twitter_handle", "twitter_name", "measure_subtitle", "measure_text", "measure_title", "office_name", "first_name", "middle_name", "last_name", "party", "organization_name", "organization_twitter_handle", "twitter_description" ] } }}

Basically, you just need to add the "type": "phrase_prefix" to the query.

**Comment by josephevans** [comment]
Once the code is updated to use a "phrase_prefix" type query, I think we can close this since it should return all results that contain the search term -- ie "Planned Parenthood" should be found if you search for "plan" or even "p".

**Comment by DaleMcGrew** [comment]
This worked, thank you!
